pywren/wrenhandler.py

```diff
@@ -92,14 +93,13 @@ def download_runtime_if_necessary(s3_client, runtime_s3_bucket, runtime_s3_key):
 
     res = s3_client.get_object(Bucket=runtime_s3_bucket,
                                Key=runtime_s3_key)
-
+    res_buffer = res['Body'].read()
     try:
-
-        condatar = tarfile.open(
-            mode="r:gz",
-            fileobj=wrenutil.WrappedStreamingBody(res['Body'],
-                                                  res['ContentLength']))
+        res_buffer_io = io.BytesIO(res_buffer)
+        condatar = tarfile.open(mode="r:gz", fileobj=res_buffer_io)
         condatar.extractall(runtime_etag_dir)
+        del res_buffer_io # attempt to clear this proactively
+        del res_buffer
```

Comment:
I think `del` does not immediately clear up memory. We can enforce gc by calling system.gc()